src/DynamoCoreWpf/UI/InOutPortPanel.cs

```diff
@@ -48,8 +52,23 @@ protected override Size MeasureOverride(Size constraint)
                     // Default behavior of getting each child to measure.
                     child.Measure(constraint);
                 }
-                catch(System.Exception)
+                catch(XamlParseException e)
                 {
+                    if(e != null)
+                        //if the inner exeption is not null, gather its inner stack trace
+                        //and create a new exception 
+                    {   if(e.InnerException != null)
+                        {
+                            var aggregatedException = new DynamoUtilities.ExceptionHelpers.DynamoWrappedException
+                                (e,
+                                $"{e.StackTrace} {System.Environment.NewLine} {e.InnerException.StackTrace}" );
+
+                             Analytics.TrackException(aggregatedException,false);
+                            continue;
```

Comment:
What is the `continue` for? Did you intend the outer `if` to be a `while(e != null)`? Can't we simply aggregate all innerexceptions into a single exception and track it just once in analytics?

Reply:
I left the continue from @QilongTang 's original change, but I like your idea better - can try that.

Reply:
whoops, but I forgot to move it 🤦 - see line 72.